velox/dwio/CMakeLists.txt

```diff
@@ -35,5 +35,29 @@ add_subdirectory(common)
 add_subdirectory(catalog)
 add_subdirectory(dwrf)
 add_subdirectory(orc)
-add_subdirectory(parquet)
 add_subdirectory(text)
+
+if(VELOX_ENABLE_PARQUET)
+  add_subdirectory(parquet)
+endif()
+
+add_library(velox_dwio_registered_readers STATIC RegisterReaders.cpp)
```

Comment:
Why do you force it to be static?

Reply:
Sorry, not sure how I missed this before but this should be part of the main library as it's not a utility for tests or fuzzers, so please use our wrapper functions:
- `velox_add_library` without any library type keyword 
- `velox_link_libraries` instead of `target_link_libraries`. 

Sorry, this isn't well documented yet (#14941).